actionview/lib/action_view/helpers/date_helper.rb

```diff
@@ -887,7 +887,7 @@ def set_day_if_discarded
         # name array has a leading +nil+ element.
         def month_names
           @month_names ||= begin
-            month_names = @options[:use_month_names] || translated_month_names
+            month_names = @options[:use_month_names] || translated_month_names.dup
             month_names.unshift(nil) if month_names.size < 13
```

Comment:
```suggestion
            month_names = @options[:use_month_names] || translated_month_names
            month_names.dup.unshift(nil) if month_names.size < 13
```

What do you think about this? Then the array is only duplicated if it needs to be modified

(considering this method gets memoized this difference is probably negligible though)

Reply:
Once I committed the suggestion, it did not work as expected, so I reverted it.

Reply:
@hirotaka I believe skipkayhil meant `month_names = month_names.dup.unshift(nil)`.

The `dup` result must be assigned back to `month_names`, otherwise subsequent code cannot access it.

I think this is a good idea, because it also avoids mutating `@options[:use_month_names]` (which comes from the user and could also be a frozen array).

Although, in microbenchmarks, I have found that `unshift` is a bit slower than splatting the array:

```suggestion
            month_names = @options[:use_month_names] || translated_month_names
            month_names = [nil, *month_names] if month_names.size < 13
```

Reply:
Thanks for the suggestion 😃